app/src/main/java/com/muzima/adapters/cohort/CohortRecyclerViewAdapter.java

```diff
@@ -0,0 +1,113 @@
+package com.muzima.adapters.cohort;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.muzima.MuzimaApplication;
+import com.muzima.R;
+import com.muzima.adapters.tags.TagsAdapter;
+import com.muzima.api.model.Tag;
+import com.muzima.model.cohort.CohortItem;
+import com.muzima.utils.MuzimaPreferences;
+
+import java.util.List;
+import java.util.Locale;
+
+public class CohortRecyclerViewAdapter extends RecyclerView.Adapter<CohortRecyclerViewAdapter.ViewHolder> {
+
+    private Context context;
+    private List<CohortItem> cohortList;
+    private OnCohortClickedListener cohortClickedListener;
+
+    public CohortRecyclerViewAdapter(Context context, List<CohortItem> cohortList, OnCohortClickedListener cohortClickedListener) {
+        this.context = context;
+        this.cohortList = cohortList;
+        this.cohortClickedListener = cohortClickedListener;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_cohort_layout, parent, false);
+        return new ViewHolder(view, cohortClickedListener);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull CohortRecyclerViewAdapter.ViewHolder holder, int position) {
+        CohortItem cohortItem = cohortList.get(position);
+        holder.titleTextView.setText(cohortItem.getCohort().getName());
+        holder.clientsCountTextView.setText(String.format(Locale.getDefault(), "%s Clients", String.valueOf(cohortItem.getCohort().getSize())));
+        if (((MuzimaApplication) context.getApplicationContext()).getCohortController()
+                .isDownloaded(cohortItem.getCohort())) {
+            if (((MuzimaApplication) context.getApplicationContext()).getCohortController()
+                    .isDownloaded(cohortItem.getCohort()) && cohortItem.getCohort().getSyncStatus() != 1)
```

Comment:
@samuelowino did you intend to call getSyncStatus()-(checks if the cohort is downloaded for both empty and non empty cohorts) or isUpdateAvailable() - checks if a cohort has pending changes to download)

Reply:
Hey @benamonya I actually did not know about the existence of this function hence the call to getSyncStatus(). Thanks for sharing this, I already updated the block to call getSyncStatus() I think that makes the code read better and easy to debug.